### PR TITLE
Add optimized hyperparameters to model info in db; format hyperparam display

### LINKED ISSUE #133: Format hyper-parameter information in models tab

### PR DESCRIPTION
**Pull request by acrellin**
Closes #133

### CODE REVIEW

**Comment by acrellin** [comment]
Been meaning to do this for a long time. Any thoughts @stefanv, @bnaul?

**Comment by acrellin** [comment]
Closes #133

**Comment by acrellin** [comment]
Here's what the models table looks like now:
image

File: cesium_app/handlers/model.py [hunk 1]
```diff
@@ -107,6 +110,8 @@ def post(self):
             params_to_optimize=params_to_optimize)
         score_future = executor.submit(build_model.score_model, computed_model,
                                        imputed_fset)
+        best_params = executor.submit(lambda cmp_mod: cmp_mod.best_params_,
+                                      computed_model) if params_to_optimize else {}
```
Comment (bnaul) [inline cesium_app/handlers/model.py]: Is it ok that this can be either a `future` or a `dict`?

**Comment by acrellin** [comment]
What do you mean by ok? The check that happens later will treat each case
appropriately.

**Comment by stefanv** [comment]
I don't understand the best_param stuff; could you explain it to me?

**Comment by acrellin** [comment]
@stefanv those are the hyperparameters that were optimized (the set of best parameters from the grid of parameters that were specified to try), which aren't available until after the model has been computed, thus all the future business.

File: cesium_app/handlers/model.py (hunk 1 above)
Comment (stefanv) [reply cesium_app/handlers/model.py]: It's kind of weird for _await_model not to actually await the model.  So why not pass the computed model as a parameter to _await_model, await it, and then grab it's optimal parameters?

**Comment by acrellin** [comment]
Yeah it is kind of weird, I wonder who wrote that in the first place 🤔😉

File: cesium_app/handlers/model.py (hunk 1 above)
Comment (bnaul) [reply cesium_app/handlers/model.py]: Maybe remove the `impute` call and just assume that the featureset was properly imputed before?

**Comment by acrellin** [comment]
I think I've addressed everyone's comments so far - anything else on this @bnaul @stefanv?

**Comment by bnaul** [comment]
Looks much nicer to me! 👍

File: cesium_app/handlers/model.py [hunk 2]
```diff
@@ -20,6 +20,54 @@
...
+
+    Returns
+    -------
+    (float, dict) tuple
+        Tuple containing (train_score, best_params) where train_score is the
+        model's training accuracy and best_params is a dictionary of best
+        hyperparameter values determined by `scikit-learn`'s `GridSearchCV`.
+        If no hyperparameter optimization is performed (i.e. `params_to_optimize`
+        is None or is an empty dict, this will be an empty dict.
+'''
```
Comment (stefanv) [inline cesium_app/handlers/model.py]: Indentation

File: cesium_app/handlers/model.py [hunk 3]
```diff
@@ -20,6 +20,54 @@
...
+        `{'alpha': [1, 2], 'beta': [4, 5, 6]}` would fit models on all
+        6 combinations of alpha and beta and compare the resulting models'
+        goodness-of-fit. If None, only those hyperparameters specified in
+        `model_parameters` will be used (passed to model constructor as-is).
+    model_path : str
+        Path indicating where serialized model will be saved.
+
+    Returns
+    -------
+    (float, dict) tuple
```
Comment (stefanv) [inline cesium_app/handlers/model.py]: Split out this list, as you would for input parameters

**Review by stefanv — COMMENTED** [review]
Awesome, thanks for the refactor!

**Comment by acrellin** [comment]
@stefanv thanks for the feedback, will do. Good to go after those two
things?

**Pull request merged**